platform/ios/src/MGLAnnotationView.m

```diff
@@ -27,6 +27,13 @@ - (void)prepareForReuse
     // Intentionally left blank. The default implementation of this method does nothing.
 }
 
+- (void)setCenter:(CGPoint)center
+{
+    center.x += _centerOffset.x;
+    center.y += _centerOffset.y;
```

Comment:
Setting `centerOffset` after adding it to the map view should immediately move the view rather than waiting until the next call to `-setCenter:`. A custom `-setCenterOffset:` would do the trick.